### PR TITLE
Normalize schema and protocol in create method of permission-grant

### LINKED ISSUE #530: Normalize Schema and Protocol properties within a PermissionsGrant
### Background:
When dealing with `protocol` and `schema` properties across various Interface Methods we normalize the properties into URLs.
Currently the `PermissionsGrant` interface does not normalize this within the provided `PermissionsScope`.

### Task Details:
Normalize the `schema` and `protocol` properties within `PermissionsScope` within the `create` method of the `PermissionsGrant` interface class.

## Picking Up This Issue:
- If you'd like to work on this, please comment "picking this up" below, and I'll assign the issue to you 

## Questions:
- If you need assistance or clarification, feel free to comment below.
- Feeling overwhelmed? Join our [`#hack-together` channel](https://discord.gg/tbd) and collaborate with a buddy.
- New to open-source? Check out our [Hacktober `#getting-started` channel](https://discord.gg/tbd).
- [Leaderboard](https://github.com/TBD54566975/developer.tbd.website/issues/721): Check your ranking after finishing this task.

### Resources:
- **Creating a Pull Request**: If you're new to GitHub and unsure how to create a pull request, follow this [step-by-step guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request).

Remember, communication is key! If you have any questions or face any challenges, we're here to help so please don't hesitate to reach out.

Good Luck! 🍁

### PR DESCRIPTION
**Pull request by adarsh-a-tw**
Fixes #530 

1. Normalize `schema` and `protocol`  properties of `PermissionScope` object in `create` method of `PermissionGrant` interface.
2. Added tests to validate normalization.

### CODE REVIEW

File: tests/interfaces/permissions-grant.spec.ts [hunk 1]
```diff
@@ -35,6 +36,56 @@ describe('PermissionsGrant', () => {
...
+      });
+
+      it('ensures that `protocol` is normalized', async () => {
+        const { privateJwk } = await Secp256k1.generateKeyPair();
+        const authorizationSigner = new PrivateKeySigner({ privateJwk, keyId: 'did:jank:bob' });
+
+        const scope: PermissionScope = {
+          interface : DwnInterfaceName.Records,
+          method    : DwnMethodName.Write,
+          protocol  : 'https://example.com/',
```
Comment (LiranCohen) [inline tests/interfaces/permissions-grant.spec.ts:71]: nit: keeps tests consistent and gives better assurance of normalization.

```suggestion
          protocol  : 'example.com/',
```

**Comment by LiranCohen** [comment]
@adarsh-a-tw looks great! Just a small nit on the tests.  🎸

File: tests/interfaces/permissions-grant.spec.ts (hunk 1 above)
Comment (adarsh-a-tw) [reply tests/interfaces/permissions-grant.spec.ts]: @LiranCohen Thanks for the suggestion! Have made the change.

**Review by LiranCohen — DISMISSED** [review]
Awesome!

**Review by LiranCohen — DISMISSED** [review]

**Review by diehuxx — DISMISSED** [review]

**Review by thehenrytsai — DISMISSED** [review]

**Review by diehuxx — APPROVED** [review]

**Pull request merged**